client/src/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import $ from 'jquery';
+import styled from 'styled-components';
+import Product from './components/Product.jsx';
+import Styles from './components/Styles.jsx';
+
+const ProductInfo = styled.div`
+  display: flex;
+  flex-flow: column;
+`;
+
+class App extends React.Component {
+
+  constructor(props) {
+    const ids = window.location.pathname.split('/');
+    super(props);
+    this.state = {
+      productId: ids[2],
+      styleId: ids[3],
+      name: '',
+      gender: '',
+      category: '',
+      style: {},
+      styles: [],
+    };
+  }
+
+  setThumbnails(styles, photos) {
+    let stylesUpdated = [];
+    styles.map(style => {
+      photos.map(photo => {
+        if (photo.style_id === style.styleId) {
+          style['imageThumbnail'] = photo.main_photo.thumbnail_url;
+          style['imageRegular'] = photo.main_photo.regular_url;
+          stylesUpdated.push(style);
+        }
+      });
+    });
+    return stylesUpdated;
+  }
+
+  componentDidMount() {
+    $.ajax({
+      url: `http://localhost:3008/product/${this.state.productId}`
+    }).done((data) => {
+      $.ajax({
+        url: `http://localhost:3000/photos/${this.state.productId}`
+      }).done((photos) => {
+        let stylesUpdated = this.setThumbnails(data.styles, photos);
+        console.log('with thumbnails', stylesUpdated);
+        this.setState({
+          productId: data.productId,
+          name: data.name,
+          gender: data.gender === 'male' ? 'Men\'s' : 'Women\'s',
+          category: data.category,
+          style: stylesUpdated.filter(s => this.state.styleId === s.styleId)[0],
+          styles: stylesUpdated
+        });
+      });
+    });
+  }
+
+  render() {
+    console.log('====render state', this.state);
+    return (
+      <ProductInfo>
+        <Product name = {this.state.name} gender= {this.state.gender} category = {this.state.category} price = {this.state.style.price} />
+        <Styles styles = {this.state.styles} />
+      </ProductInfo>
+    );
+  }
+}
+
+ReactDOM.render(<App />, document.getElementById('app'));
```

Comment:
Can you rename your component from "App" to something else? This is primarily because it will cause conflicts with the other components the team is working on.

For example, Martin's review module is also rendering to "app," which means when I ran my proxy one module overwrote the other.

As an example of how to change this, my photo-gallery component is now rendering to “photo-module”.

Reply:
Makes sense.  Changed.